BuildLibs/buildmap.py

```diff
@@ -1,153 +1,190 @@
+        self.shade: int = kwargs.get('shade', 0)
+        self.palette: int = kwargs.get('palette', 0)
+        self.texcoords: list = kwargs.get('texcoords', [0, 0, 0, 0])
+        self.lowtag: int = kwargs.get('lowtag', 0)
+        self.hightag: int = kwargs.get('hightag', 0)
+        self.extra: int = kwargs.get('extra', -1)
+        self.length: int = kwargs.get('length', 0)
 
 
 class Sprite:
-    def __init__(self, d):
-        self.__dict__ = d
-        self.pos:list
-        self.picnum:int
-        self.sectnum:int
-        self.length:int
 
-    def copy(self) -> 'Sprite':
-        d = self.__dict__.copy()
-        for k in d.keys():
-            if type(d[k]) != int:
-                d[k] = d[k].copy()
-        return Sprite(d)
+    def __init__(self, **kwargs):
+        self.pos: list = kwargs.get('pos', [0, 0, 0])
+        self.cstat: int = kwargs.get('cstat', 0)
+        self.picnum: int = kwargs.get('picnum', 0)
+        self.shade: int = kwargs.get('shade', 0)
+        self.palette: int = kwargs.get('palette', 0)
+        self.clipdist: int = kwargs.get('clipdist', 0)
+        self.filler: int = kwargs.get('filler', 0)
+        self.texcoords: list = kwargs.get('texcoords', [0, 0, 0, 0])
+        self.sectnum: int = kwargs.get('sectnum', 0)
+        self.statnum: int = kwargs.get('statnum', 0)
+        self.angle: int = kwargs.get('angle', 0)
+        self.owner: int = kwargs.get('owner', 0)
+        self.velocity: list = kwargs.get('velocity', [0, 0, 0])
+        self.lowtag: int = kwargs.get('lowtag', 0)
+        self.hightag: int = kwargs.get('hightag', 0)
+        self.extra: int = kwargs.get('extra', -1)
+        self.length: int = kwargs.get('length', 0)
+
+    def __copy__(self) -> 'Sprite':
+        cls = self.__class__
+        copied = cls.__new__(cls)
+        copied.__dict__.update(self.__dict__)
+        copied.pos = copied.pos[:]
+        copied.texcoords = copied.texcoords[:]
+        copied.velocity = copied.velocity[:]
+        return copied
 
-    def __repr__(self):
-        return repr(self.__dict__)
 
+class MapFile:
 
-def LoadMap(gameName, name, data: bytearray) -> 'MapFile':
-    # Blood has a signature at the front of map files
-    if data[:4] == b'BLM\x1a':
-        return BloodMap(gameName, name, data)
-    else:
-        (version,) = unpack('<i', data[:4])
-        if version <= 6:
-            # https://moddingwiki.shikadi.net/wiki/MAP_Format_(Build)#Version_6
-            return MapV6(gameName, name, data)
-        return MapFile(gameName, name, data)
+    """
+    https://moddingwiki.shikadi.net/wiki/MAP_Format_(Build)
 
+    """
 
-class MapFile:
-    # https://moddingwiki.shikadi.net/wiki/MAP_Format_(Build)
-    def __init__(self, gameName, name, data: bytearray):
-        info('\n', name, len(data))
+    HEADER_SIZE = 20
+    SECTOR_SIZE = 40
+    WALL_SIZE = 32
+    SPRITE_SIZE = 44
+    X_SECTOR_SIZE = 0
+    X_WALL_SIZE = 0
+    X_SPRITE_SIZE = 0
```

Comment:
these are unused class variables because we use them as members instead

```
            self.X_SECTOR_SIZE = 60
            self.X_SPRITE_SIZE = 56
            self.X_WALL_SIZE = 24
```